apps/src/netsim/NetSimPacketEditor.js

```diff
@@ -568,35 +625,68 @@ NetSimPacketEditor.prototype.updateFields_ = function (skipElement) {
   var chunkSize = this.currentChunkSize_;
   var liveFields = [];
 
+  var level = netsimGlobals.getLevelConfig();
+  var encoder = new Packet.Encoder(level.addressFormat,
+      level.packetCountBitWidth, this.packetSpec_);
+
   this.packetSpec_.forEach(function (fieldSpec) {
     /** @type {Packet.HeaderType} */
-    var fieldName = fieldSpec.key;
+    var fieldName = fieldSpec;
     /** @type {number} */
-    var fieldWidth = fieldSpec.bits;
+    var fieldWidth = encoder.getFieldBitWidth(fieldName);
+
+    var abConverter, binaryConverter, hexConverter, decimalConverter, asciiConverter;
+    if (Packet.isAddressField(fieldName)) {
+      abConverter = function (addressString) {
+        return dataConverters.binaryToAB(
+            dataConverters.addressStringToBinary(
+                addressString, level.addressFormat));
+      };
+      binaryConverter = function (addressString) {
+        return dataConverters.formatBinaryForAddressHeader(
+            dataConverters.addressStringToBinary(
+                addressString,
+                level.addressFormat),
+            level.addressFormat);
+      };
+      hexConverter = function (addressString) {
+        return dataConverters.binaryToHex(
+            dataConverters.addressStringToBinary(
+                addressString, level.addressFormat));
+      };
+      decimalConverter = cleanAddressString;
+      asciiConverter = cleanAddressString;
+    } else {
+      abConverter = intToAB;
+      binaryConverter = intToBinary;
+      hexConverter = intToHex;
+      decimalConverter = function (val) { return val.toString(10); };
```

Comment:
nit: should technically prob be split across lines. i dont care too much either way